pkgs/applications/editors/tijolo/default.nix

```diff
@@ -0,0 +1,58 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, crystal
+, vte
+, libgit2
+, editorconfig-core-c
+, gtksourceview4
+, wrapGAppsHook
+, desktopToDarwinBundle
+}:
+crystal.buildCrystalPackage rec {
+  pname = "tijolo";
+  version = "0.7.3";
+
+  src = fetchFromGitHub {
+    owner = "hugopl";
+    repo = "tijolo";
+    rev = "v${version}";
+    hash = "sha256-15not/B+O+wIZ/fvLFy26/dyvq0E+bZUeoSZ6HxMMKg=";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ]
+    ++ lib.optionals stdenv.isDarwin [ desktopToDarwinBundle ];
+  buildInputs = [ vte libgit2 gtksourceview4 editorconfig-core-c ];
+
+  format = "make";
+  shardsFile = ./tijolo-shards.nix;
+
+  doCheck = false;
+  doInstallCheck = false;
+
+  buildPhase = ''
+    runHook preBuild
+
+    make
```

Comment:
Why can't we use the default build phase?

Reply:
I guess that `buildPhase` runs `make build` command and fails with error "there is no build rule for makefile". So i call `make`, not `make build` and it works.

Reply:
please check if there is a builder option for that, we don't want to override the entire buildPhase as that it will also override certain flags that we might want to use.

Reply:
I didn't quite understand what builder option i need to find. I can patch `Makefile` and add `build` rule :D

Reply:
Try to explore this https://nixos.org/manual/nixpkgs/unstable/#build-phase

Reply:
I didn't find a way to override the default make rule. I think that in such cases `buildPhase` is needed

Reply:
I can simply patch `Makefile`, all we need is to replace `all` to `build`. Just one word.
```
all:
	shards build --release -s  -Dpreview_mt --ignore-crystal-version
```

Reply:
It seems there's an open issue to tackle this (with a workaround) https://github.com/NixOS/nixpkgs/issues/28047

Reply:
I figured out all problems, but now i'm stuck this
```
(crystal-run-spec.tmp:329): Gtk-CRITICAL **: 14:27:19.744: _gtk_css_provider_load_named: assertion '!g_str_equal (name, DEFAULT_THEME_NAME)' failed
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
```
it's still the check phase.
Terminal freezes, nothing happens.

Reply:
It works. Fixed in https://github.com/NixOS/nixpkgs/pull/243768/commits/53d07ea20629875d965242b98ebb168464224641

Reply:
Can you squash that into the previous commit? This PR shouldn't be more than 2 commits.

Reply:
Fixed